scripts/config.py

```diff
@@ -208,22 +220,28 @@ def full_adapter(name, active, section):
         return active
     return include_in_full(name)
 
+# The baremetal configuration excludes options that require a library or
+# operating system feature that is typically not present on bare metal
+# systems. Features that are excluded from "full" won't be in "baremetal"
+# either (unless explicitly turned on in baremetal_adapter) so they don't
+# need to be repeated here.
 EXCLUDE_FROM_BAREMETAL = frozenset([
+    #pylint: disable=line-too-long
     'MBEDTLS_DEPRECATED_WARNING',
-    'MBEDTLS_ENTROPY_NV_SEED',
-    'MBEDTLS_FS_IO',
-    'MBEDTLS_HAVEGE_C',
-    'MBEDTLS_HAVE_TIME',
-    'MBEDTLS_HAVE_TIME_DATE',
-    'MBEDTLS_NET_C',
-    'MBEDTLS_PLATFORM_FPRINTF_ALT',
-    'MBEDTLS_PLATFORM_TIME_ALT',
-    'MBEDTLS_PSA_CRYPTO_SE_C',
-    'MBEDTLS_PSA_CRYPTO_STORAGE_C',
-    'MBEDTLS_PSA_ITS_FILE_C',
-    'MBEDTLS_THREADING_C',
-    'MBEDTLS_THREADING_PTHREAD',
-    'MBEDTLS_TIMING_C',
+    'MBEDTLS_ENTROPY_NV_SEED', # requires FS_IO or alternate NV seed hooks
+    'MBEDTLS_FS_IO', # requires a filesystem
+    'MBEDTLS_HAVEGE_C', # requires a clock
+    'MBEDTLS_HAVE_TIME', # requires a clock
+    'MBEDTLS_HAVE_TIME_DATE', # requires a clock
+    'MBEDTLS_NET_C', # requires POSIX-like networking
+    'MBEDTLS_PLATFORM_FPRINTF_ALT', # requires FILE* from stdio.h
+    'MBEDTLS_PLATFORM_TIME_ALT', # requires timing
+    'MBEDTLS_PSA_CRYPTO_SE_C', # requires a filesystem
```

Comment:
It depends on `MBEDTLS_PSA_CRYPTO_STORAGE_C`, not `MBEDTLS_PSA_ITS_FILE_C`: you can link with an alternative implementation of ITS instead of `MBEDTLS_PSA_ITS_FILE_C`. But generally speaking, yes, I'll add some dependency information.